### ISSUE TITLE
[0.76] feat: Expose MetroConfig type directly from @react-native/metro-config

### ISSUE THREAD

**Issue by tjzel** [reactwg/react-native-releases#516]
### Target Branch(es)

0.76

### Link to commit or PR to be picked

https://github.com/facebook/react-native/pull/46602

### Description

Would be nice to have in stable 0.76, especially since RNC CLI isn't going to be embedded soon.

**Comment by blakef** [comment]
This needs to land in `main` before we can pick it for 0.76

**Comment by robhogan** [comment]
@blakef May be too late for today's RC but this has now landed in main ( https://github.com/facebook/react-native/commit/cc6d1eb8449c25fe3c76aaf4da5cd38b3ac0783d ) - my fault for sitting on it, apologies!
